src/inverted-index.js

```diff
@@ -1,88 +1,151 @@
+      .toLowerCase()
+      .split(' ')
+      .sort();
+    return (cleanText);
+  }
 
-class InvertedIndex{
-	constructor(){
-		//Object to store the index
-		this.index = {};
-	}
+  /**
+   * @param {Array} words - An array of strings
+   * @return {Array} uniqueWords - An array non-duplicate strings
+   */
+  static removeDuplicates(words) {
+    const uniqueWords = words.filter((item, index) => words.indexOf(item) === index);
+    return (uniqueWords);
+  }
 
-	/*
-	* Function to separate a string into words
-	* Non-alphanumeric characters are removed 
-	* The text is converted to lower case and sorted
-	*/
-	static clean(text){
-		let cleanText = text.replace(/[.,\/#!$%\^&\*;:{}=\-_`~()]/g,"")
-		 					.replace(/\s{2,}/g, " ")
-		 					.toLowerCase()
-		 					.split(' ')
-		 					.sort();
-		return(cleanText);
-	}
+  /**
+   * @param {Object} index - Efizi function to sort an Object
+   * @return {Object} sortedObject - Object with keys sorted
+   */
+  static sortObjectKeys(index) {
+    const sortedKeys = Object.keys(index).sort();
+    // Object that will contain the sorted object
+    const sortedObject = {};
+    sortedKeys.forEach((key) => {
+      sortedObject[key] = index[key];
+    });
+    return (sortedObject);
+  }
 
-	/*
-	* Function to remove duplicate words
-	* from an array of strings
-	*/
+  /**
+   * @param{String} filename - Name of the file for which index is to be created
+   * @param{Object} docToIndex - JSON file
+   * @returns{undefined} - This function does not return anything it just
+   * updates the this.indexes Object
+   */
+  createIndex(filename, docToIndex) {
+    // This object stores the index of the current document
+    let index = {};
+    // Ensure JSON file is not empty
+    if (docToIndex.length === 0) {
+      this.indexes[filename] = 'JSON file is empty';
+    } else if (Object.keys(docToIndex[0]).indexOf('title') === -1 || Object.keys(docToIndex[0]).indexOf('text') === -1) {
+      this.indexes[filename] = 'JSON file is invalid';
+    } else {
+      docToIndex.forEach((document, docIndex) => {
+        const cleanWords = InvertedIndex.clean(document.text);
+        const uniqueWords = InvertedIndex.removeDuplicates(cleanWords);
 
-	static removeDuplicates(words){
-		let uniqueWords = words.filter((item, index) => words.indexOf(item) === index);
-		return(uniqueWords);
-	}
+        uniqueWords.forEach((word) => {
+          if (Object.keys(index).indexOf(word) === -1) {
+            index[word] = [];
+            index[word].push(docIndex + 1);
+          } else {
+            index[word].push(docIndex + 1);
+          }
+        });
+      });
+      // Sort the index keys
+      index = InvertedIndex.sortObjectKeys(index);
+      // Simply update the this.indexes object
+      // Don't return it yet
+      this.indexes[filename] = index;
+    }
+  }
 
-	getIndex(docToIndex){
-		const index = {};
+  /**
+   * @returns{Object} - Object that contains all the index
+   */
+  getIndex() {
```

Comment:
According to the requirement, this method is meant to return a particular index.